### PR TITLE
Issue #2 | imported NavLink and replace anchor tags - ac & sj

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate to all the pages in the application by clicking the links in the nav bar
## Summary

Users need to be able to navigate between the pages of our application in order to access all of its features. There is a navigation bar in the `Layout.jsx` component, but it's not quite feature-complete: the links in the nav element don't actually allow the user to navigate from page to page.

## Acceptance criteria
- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Notes
- The [official docs for the `NavLink` component](https://reactrouterdotcom.fly.dev/docs/en/v6/components/nav-link) will come in very handy!
- While a standard anchor element (`<a>`) _could_ allow a user to navigate from page to page, the `NavLink` component adds additional behavior that is essential to our app functioning correctly!

### PR DESCRIPTION
**Pull request by arcan9**
### Navigate between pages using links

This code updates the `Layout.jsx` component to activate links to `Home`, `List`, and `add-item`.  We imported `NavLink` from react-router-dom and replaced the anchor tags with `NavLink` tags. We learned that `NavLink` adds an `active` class that tells us  whether a link is active which is essential for our app to work smoothly.

## Related Issue

closes #2 

## Acceptance Criteria

- [x] The `nav` element is added to the `Layout` component
- [x] Links to the "Home", "List", and "Add item" pages of the app are In the `nav` element
- [x] The links all function as expected using  the `NavLink` component from `react-router-dom`

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Screen Shot 2023-01-12 at 4 04 56 PM

### After

Screenshot 2023-01-12 150508

## Testing Steps / QA Criteria
- From your terminal, pull down this branch with git pull origin `ac-sj-shopping-list-navigation` and check that branch out with git checkout  `ac-sj-shopping-list-navigation`
- From your terminal, type `npm start` to open http://localhost:3000
- Click on links in navigation to go to the different pages

### CODE REVIEW

**Review by Bustamor — COMMENTED** [review]
After reading through and clicking the preview, everything looks solid.  Nice work both of you!

**Review by alex-andria — APPROVED** [review]
Great work @arcan9 and @sarahmjenkins in updating the navigation bar with `NavLink` and making it functional!

File: src/views/Layout.jsx [hunk 1]
```diff
@@ -1,15 +1,7 @@
-import { Outlet } from 'react-router-dom';
+import { Outlet, NavLink } from 'react-router-dom';
 
 import './Layout.css';
 
-/**
```
Comment (mtfoley) [inline src/views/Layout.jsx:5]: Glad to see you removing the TODO comment! ✅

**Review by mtfoley — APPROVED** [review]
Changes look great @arcan9 and @sarahmjenkins!

File: src/views/Layout.jsx [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
- * TODO: The links defined in this file don't work!
- *
- * Instead of anchor element, they should use a component
- * from `react-router-dom` to navigate to the routes
- * defined in `App.jsx`.
- */
-
 export function Layout() {
```
Comment (djtaylor8) [inline src/views/Layout.jsx:5]: Maybe add a short comment about what this component does. As you build out more components and add features, adding simple documentation will help you and others (especially in the future!) have a clear idea of why this component exists. I realize it seems simple for a basic component like this, but it's a good habit to get into moving forward!

Example: 
`/** Layout component that display navigation links in the UI */`

**Review by djtaylor8 — APPROVED** [review]
Way to go you two! As @mtfoley mentioned, thank you for cleaning up the TODO comments. Everything is working as expected. Great work!

**Review by daredtech — APPROVED** [review]
Great job, team! 🚀

**Pull request merged**